shared/profile/wot-author/index.tsx

```diff
@@ -3,59 +3,122 @@ import {WebOfTrustVerificationType} from '../../constants/types/more'
 }
 
-// PICNIC-1059 Keep in sync with server limit (yet to be implemented)
-const statementLimit = 700
-
 const WotModal = (props: WotModalProps) => {
+  const dispatch = Container.useDispatch()
+  const onClose = () => dispatch(RouteTreeGen.createClearModals())
   return (
     <Kb.Modal
-      header={{title: 'Web of Trust'}}
+      onClose={onClose}
+      scrollViewRef={props.scrollViewRef}
+      header={{
+        leftButton: props.onBack ? (
+          <Kb.Text onClick={props.onBack} type="BodyPrimaryLink">
+            Back
+          </Kb.Text>
+        ) : Styles.isMobile ? (
+          <Kb.Text onClick={onClose} type="BodyPrimaryLink">
+            Cancel
+          </Kb.Text>
+        ) : (
+          undefined
+        ),
+        title: 'Web of Trust',
+      }}
       mode="DefaultFullHeight"
       banners={[
         !!props.error && (
-          <Kb.Banner color="red">
+          <Kb.Banner key="error" color="red">
             <Kb.BannerParagraph bannerColor="red" content={props.error} />
           </Kb.Banner>
         ),
       ]}
       footer={{
         content: (
           <Kb.ButtonBar align="center" direction="row" fullWidth={true} style={styles.buttonBar}>
-            <Kb.Button fullWidth={true} label={props.submitLabel} onClick={props.onSubmit} />
+            <Kb.Button
+              fullWidth={true}
+              label={props.submitLabel}
+              onClick={props.onSubmit}
+              disabled={props.submitDisabled}
+              waiting={props.submitWaiting}
+            />
           </Kb.ButtonBar>
         ),
       }}
     >
-      <Kb.Box2
-        direction="vertical"
-        alignSelf="stretch"
-        alignItems="stretch"
-        style={{backgroundColor: Styles.globalColors.green}}
-      >
+      <Kb.Box2 direction="vertical" alignSelf="stretch" alignItems="center" style={styles.topIconContainer}>
         <Kb.Icon type="icon-illustration-wot-460-96" />
       </Kb.Box2>
       {props.children}
     </Kb.Modal>
   )
 }
 
-export const Question1 = (props: Props) => {
+type Question1Props = {
+  error?: string
+  initialVerificationType: WebOfTrustVerificationType
+  onSubmit: (_: {
+    otherText: string
+    proofs: {key: string; value: string}[]
+    verificationType: WebOfTrustVerificationType
+  }) => void
+  proofs: {key: string; value: string}[]
+  voucheeUsername: string
+}
+
+export const Question1 = (props: Question1Props) => {
+  const scrollViewRef = React.useRef<Kb.ScrollView>(null)
+  const [selectedVt, _setVt] = React.useState<WebOfTrustVerificationType>(props.initialVerificationType)
+  const [otherText, setOtherText] = React.useState('')
+  const setVt = newVt => {
+    if (newVt === 'other' && newVt !== selectedVt && scrollViewRef.current) {
+      const hackDelay = 50 // With no delay scrolling undershoots. Perhaps the bottom component doesn't exist yet.
+      setTimeout(() => scrollViewRef.current?.scrollToEnd({animated: true}), hackDelay)
+    }
+    _setVt(newVt)
+  }
```

Comment:
Why the fear of long variable names here? `selectedVerificationType` and `setVerificationType` @mlsteele